tags_test.go

```diff
@@ -8,6 +8,105 @@ import (
 	"github.com/rocktavious/autopilot/v2023"
 )
 
+type extractTagsTestCase struct {
+	tagsWanted           []ol.Tag
+	existingTags         []ol.Tag
+	expectedTagsToCreate []ol.Tag
+	expectedTagsToDelete []ol.Tag
+}
+
+func TestExtractTags(t *testing.T) {
+	var noTags []ol.Tag
+	fourTags := []ol.Tag{
+		{Key: "foo", Value: "bar"},
+		{Key: "ping", Value: "pong"},
+		{Key: "marco", Value: "pollo"},
+		{Key: "env", Value: "prod"},
+	}
+	// Arrange
+	testCases := map[string]extractTagsTestCase{
+		"create all delete none": {
+			tagsWanted:           fourTags,
+			existingTags:         noTags,
+			expectedTagsToCreate: fourTags,
+			expectedTagsToDelete: noTags,
+		},
+		"create none delete all": {
+			tagsWanted:           noTags,
+			existingTags:         fourTags,
+			expectedTagsToCreate: noTags,
+			expectedTagsToDelete: fourTags,
+		},
+		"create some delete some": {
+			tagsWanted:           fourTags[:3],
```

Comment:
IMO these tests are hard to read because of slice notation and also because I have to look up what `fourTags` is by scrolling up. I would prefer we just wrote the literal array in place of `noTags` and `fourTags`. Same for aliases_test.go.

(This is a non-blocking suggestion.)